numpy/random/_generator.pyx

```diff
@@ -3535,7 +3535,7 @@ cdef class Generator:
             # if check_valid == 'ignore' we need to ensure that np.sqrt does not
             # return a NaN if s is a very small negative number that is
             # approximately zero or when the covariance is not positive-semidefinite
-            _factor = u * np.sqrt(abs(s))
+            _factor = np.transpose(u * np.sqrt(abs(s)))
```

Comment:
I am a bit worried that we silently accept more than 2 dimensions here (what happens for more than 2 dimensions? If it is an error, just having a test for that could be fine as well, maybe with a comment that the code may need updating), but than transpose would have to transpose only the last two dimensions, maybe `np.swapaxes` is actually good (for once).

The main point is, we really need at least some test, a random/fuzzy one that is lenient enough to not fail would seem fine to me. Or maybe one that checks approximate consistency over the different `methods` at least?

Reply:
Agree, we can be explicit about transposing the last two dimensions. Also agree on the tests, will add some this weekend.

Reply:
> but than transpose would have to transpose only the last two dimensions

That is what I meant about the standard form being easier to extend. But I think we can deal with that larger refactoring if we ever decide to go the stacked route.

Reply:
> but than transpose would have to transpose only the last two dimensions

The pseido-private `linalg.transpose` does this.